lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -514,14 +757,15 @@ class HandshakeToHWPass : public HandshakeToHWBase<HandshakeToHWPass> {
     // graph. This ensures that any referenced submodules (through
     // handshake.instance) has already been lowered, and their HW module
     // equivalents are available.
+    OpBuilder submoduleBuilder(mod.getContext());
```

Comment:
Is the post-order traversal necessary, or could one extract the symbol name from the `InstanceOp`s?

Reply:
In theory, yes. But I think the code turns out more clean this way. Ensuring that whatever an `InstanceOp` is referencing has always already been lowered ensures that we can just lookup that symbol when needed. Assuming that we find the call to `resolveInstanceGraph` a mandatory correctness check, this post-order iteration of modules doesn't add any additional overhead either.

Reply:
Shouldn't `resolveInstanceGraph` by checked by a verifier, or is there handshake code that verifies but doesn't satisfy this precondition?

Reply:
I'm not sure - If a verifier checks this, it is in other words saying that recursive `handshake` functions are illegal. They are illegal in a HW lowering sense, but i do not think/know whether this should be the case for the abstraction itself.

Reply:
Then let's keep it as is for now. If we later on figure that this should be checked in a verifier, we can still change things.